controllers/admin/AdminEmailsController.php

```diff
@@ -235,7 +235,12 @@ public function setMedia()
 	
 	public function processDelete()
 	{
-		return Mail::eraseAllLogs();
+		if ((int)$id_mail = Tools::getValue('id_mail', 0))
```

Comment:
It will ; I'd tested.

Default value is to 0, so the if is false statement and do the eraseAll().